pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -564,6 +565,13 @@ type GameServerMetadata struct {
 
 // ApplyDefaults applies the default values to this GameServerAllocation
 func (gsa *GameServerAllocation) ApplyDefaults() {
+	if len(gsa.Spec.Priorities) == 0 {
```

Comment:
Please add tests to:

https://github.com/googleforgames/agones/blob/a1a1cd956a19e4e97fd67f6c0ad06da4259273ab/pkg/apis/allocation/v1/gameserverallocation_test.go#L31